src/librustc_plugin/deprecated/Cargo.toml

```diff
@@ -0,0 +1,14 @@
+[package]
+authors = ["The Rust Project Developers"]
+name = "rustc_plugin"
+version = "0.0.0"
+build = false
+edition = "2018"
+
+[lib]
+name = "rustc_plugin"
+path = "lib.rs"
+doctest = false
+
+[dependencies]
+rustc_plugin_ = { path = ".." }
```

Comment:
I don't think this compatibility shim is necessary.
`rustc_plugin` is almost unused at this point, so it should be ok to make people to migrate to `rustc_driver::plugin` in one step.

Reply:
That’s fair, but now that the shim is already implemented it’s the same amount of work to remove it now or later.

Reply:
If this is merged now, I'll have to make a PR removing it myself once this PR lands, I don't want to do that + I'd prefer some pointless back and forth to not exist in git history.

Reply:
Ah, I see, Cargo uses it, and we cannot land this PR with broken cargo.
Ok, let's land this with the shim.

Reply:
Breaking `extern crate rustc_plugin;` in the same PR as we make its replacement available means that Clippy and RLS will be broken for a few Nightlies until we can get the multi-repo coordination dance together. This doesn’t sound great when it’s easily avoided.

If you’d like I can be on the hook for making the follow-up PR that removes the deprecated shim later.

Reply:
>If you’d like I can be on the hook for making the follow-up PR that removes the deprecated shim later.

Yeah, that would be great.